index.js

```diff
@@ -292,26 +292,33 @@ function fastifyMultipart (fastify, options = {}, done) {
 
     this.log.debug('starting multipart parsing')
 
-    let worker
-    let lastValue
+    let values = []
+    let pendingHandler = null
 
     // only one file / field can be processed at a time
     // "null" will close the consumer side
     const ch = (val) => {
-      if (typeof val === 'function') {
-        worker = val
+      if (pendingHandler) {
+        pendingHandler(val)
+        pendingHandler = null
       } else {
-        lastValue = val
+        values.push(val)
       }
-      if (worker && lastValue !== undefined) {
-        worker(lastValue)
-        worker = undefined
-        lastValue = undefined
+    }
+
+    const handle = (handler) => {
+      if (values.length > 0) {
+        const value = values[0]
+        values = values.slice(1)
```

Comment:
Why not use `handler(values.shift())` ?

Reply:
Just want to make the logic more explicit. The same reason that I use function *cb* for values and *handle* for callback function instead of using one function *cb* to handle both values and callback function.